components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/OAuth2JWTTokenValidator.java

```diff
@@ -322,6 +331,42 @@ private String getTenantDomain() {
         return tenantDomain;
     }
 
+    private String getSigningTenantDomain(JWTClaimsSet claimsSet, AccessTokenDO accessTokenDO)
```

Comment:
better to have a validation to check if accessTokenDO is null and have a fallback mechanism to call the previous getTenantDomain() method in such cases to avoid any possible NPEs

Reply:
Added a null check for accessTokenDO.